Content.Server/Chat/V2/Systems/RadioChannelValidationSystem.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Content.Shared.Chat.V2;
+using Content.Shared.Chat.V2.Components;
+using Content.Shared.Radio;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Chat.V2.Systems;
+
+public sealed class RadioChannelValidationSystem : EntitySystem
+{
+    private const string RadioChannelFailed = "chat-system-radio-channel-failed";
+    private const string RadioChannelKey = "channel";
+
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ChatValidationEvent<AttemptEquipmentRadioEvent>>((msg, args) => HandleRadioValidationEvent(msg, args));
+        SubscribeLocalEvent<ChatValidationEvent<AttemptInternalRadioEvent>>((msg, args) => HandleRadioValidationEvent(msg, args));
+    }
+
+    private void HandleRadioValidationEvent(ChatValidationEvent<AttemptEquipmentRadioEvent> msg, EntitySessionEventArgs args)
+    {
+        if (!_proto.TryIndex<RadioChannelPrototype>(msg.Event.Channel, out _))
```

Comment:
All three blocks have similar code inside - maybe join ifs? or at least invert them? A bit of a stylistic thing, but inverted if ususally is considered as 'less readable'. Also undersanding the fact that cancel messages are all the same tooke some time for me - thats a bad sign. Same thing with next method.